File: drouter-api/src/main/java/com/didi/drouter/annotation/Interceptor.java
package com.didi.drouter.annotation;

import com.didi.drouter.api.Extend;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by gaowei on 2018/9/3
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface Interceptor {

    // from large to small
    int priority() default 0;

    boolean global() default false;

    @Extend.Cache int cache() default Extend.Cache.NO;
}

File: plugins/drouter/src/main/kotlin/com/didi/drouter/annotation/Interceptor.kt
package com.didi.drouter.annotation

/**
 * Created by gaowei on 2018/9/3
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.BINARY)
annotation class Interceptor(
    val name: String = "",
    val priority: Int = 0,
    val global: Boolean = false,
    val cache: Int = 0
)
